file.go

```diff
@@ -1,37 +1,174 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	Data map[string]interface{}
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open JSON file
+	jsonFile, err := os.Open(path)
+
+	//if an error occurred
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	//defer the closing of file till the function ends
+	defer jsonFile.Close()
+
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	//if an error occurred while reading JSON file
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	//create file object
+	var myFile file
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+	json.Unmarshal(byteValue, &myFile.Data)
+
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	//create file object
+	var myFile file
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(f)
+
+	//if an error occurred while reading JSON file
+	if err != nil {
+		fmt.Println(err)
+		return file{}, nil
+	}
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+
+	json.Unmarshal(byteValue, &myFile)
+
+	return myFile, err
```

Comment:
```suggestion
	// reading input file
	byteValue, err := ioutil.ReadAll(f)
	if err != nil {
		return file{}, nil
	}
	
	data := make(map[string]interface{})
	
	// parsing file contents as json
	err = json.Unmarshal(byteValue, &data)
	if err != nil {
		return file{}, err
	}
	
	return file{data: data}, err
```

Reply:
The above will work for a `JSON` object, but not for an array.
You will need `[]map[string]interface{}` instead of `map[string]interface{}`.

Reply:
all right, I will update the code.